src/backend/cdb/cdbllize.c

```diff
@@ -863,6 +865,19 @@ fix_outer_query_motions_mutator(Node *node, decorate_subplans_with_motions_conte
 			SubPlan	   *spexpr = (SubPlan *) node;
 			decorate_subplan_info *sstate = &context->subplans[spexpr->plan_id];
 
+			if (spexpr->is_initplan)
+			{
+				ListCell   *l;
+
+				foreach(l, root->init_plans)
```

Comment:
In the assert above, pointer to `root` is checked for NULL. May it be NULL here too?

Reply:
The assert above just makes sure that the `root` exists and its type is `PlannerInfo`.